src/app/routes/connector-ui/contract-agreement-page/contract-agreement-page/contract-agreement-page.service.ts

```diff
@@ -16,8 +18,11 @@ export class ContractAgreementPageService {
   constructor(
     private edcApiService: EdcApiService,
     private contractAgreementCardMappedService: ContractAgreementCardMappedService,
+    private activeFeatureSet: ActiveFeatureSet,
   ) {}
 
+  private consumingContractAgreements: ContractAgreementCardMapped[] = [];
```

Comment:
these services are meant to be stateless, so adding fields like this is dirty code.

the correct way is to pass it as parameter, e.g.

```ts
let agreements = ...;
agreements = this.withEnforcedLimits(agreements, connectorLimits);
```